### PR TITLE
Delete wasn't setting contents to null, exists was checking for null;…

### PR DESCRIPTION
**Pull request by nquinn721**
… Updated dele

te method to set contents to null

### CODE REVIEW

**Comment by SBoudrias** [comment]
I worry this might have undesired side-effect on dependents.

Looks to me like this logic state is just wrong here: https://github.com/SBoudrias/mem-fs-editor/blob/master/lib/actions/exists.js#L6

**Comment by nquinn721** [comment]
You mentioned not wanting to change the `exists` method. I can update that method to correctly identify a deleted file if you'd like.

**Comment by SBoudrias** [comment]
I believe that'd be a better solution yeah!

**Comment by nquinn721** [comment]
So the ci test that failed doesn't seam to have anything to do with the code I updated.

File: lib/actions/exists.js [hunk 1]
```diff
@@ -3,5 +3,5 @@
 module.exports = function (filepath) {
 	var file = this.store.get(filepath);
 
-	return file.contents !== null && file.state !== 'deleted';
+	return file.contents !== null && file.contents.toString("utf-8") !== "" && file.state !== 'deleted';
```
Comment (SBoudrias) [inline lib/actions/exists.js]: It's possible someone will write an empty file.

**Comment by SBoudrias** [comment]
Yeah, looks like travis flaked.

Can you add a unit test that's proving your fix is working?

**Comment by nquinn721** [comment]
I wrote a unit test. The unit test you have for exists was throwing a false positive, yet still with the new code I wrote it would get the same outcome (this time a true positive). The unit test I wrote was more of a verification of the deep nature of what is happening.

**Comment by nquinn721** [comment]
I'm hoping to get this resolved soon since we use your system in our development.

File: test/exists.js [hunk 2]
```diff
@@ -28,4 +28,13 @@ describe('#exists()', function () {
     this.fs.delete(fileDelete);
     assert.equal(this.fs.exists(fileDelete), false);
   });
+
+  it('file contents are not null but are empty after delete', function() {
```
Comment (SBoudrias) [inline test/exists.js]: This test is never checking if the `exists` method is working or not.

File: test/exists.js [hunk 3]
```diff
@@ -28,4 +28,13 @@ describe('#exists()', function () {
     this.fs.delete(fileDelete);
```
Comment (SBoudrias) [inline test/exists.js]: This test here is testing the `exists` return false after a delete operation.

How come is it passing if you're reporting that a deletion doesn't get caught by exists?

**Comment by nquinn721** [comment]
Actually I just figured something out. The delete method needs to set content to null. The `commit` method sets state to `undefined` so the `exists` method will see that the state is `undefined` and the contents are not `null` thus saying that the file exists but the file still has empty contents. With this the `readJSON` method will use the `read` and it will return an empty string which cannot be parsed thus throwing the parse error. Also if you try to create a new one since the contents are never null the `isNew` will always be false. So the first pull request where the `delete` method sets contents to `null` is the best code option.

**Comment by blai** [comment]
This make sense, it'd be nice if you can add a test for this.

**Comment by blai** [comment]
LGTM

**Comment by SBoudrias** [comment]
Oh yeah, sorry I dropped the ball on this review. LGTM! Merging away

Thanks a lot for taking the time to go through this!

**Pull request merged**